### PR TITLE
Skip GStreamer tests if plugins are missing

### PR DESCRIPTION
**Pull request by diego-plan9**
* Add a check to ReplayGainGstCliTest that ensures that the required
initial gstreamer plugins can be loaded, skipping the test if it is not
the case instead of running it.
* Add a check to ReplayGainGstCliTest.test_cli_saves_track_gain for
checking if item.rg_track_peak and item.rg_track_gain is not None. If
they are None, it is assumed that the decoder plugins could not be
found, and the tests is skipped, as discussed on #1830.

### CODE REVIEW

File: test/test_replaygain.py [hunk 1]
```diff
@@ -132,6 +141,20 @@ def test_cli_saves_album_gain_to_file(self):
 class ReplayGainGstCliTest(ReplayGainCliTestBase, unittest.TestCase):
     backend = u'gstreamer'
 
+    def setUp(self):
+        try:
+            # Check if required plugins can be loaded by instantiating a
+            # GStreamerBackend (via its .__init__).
+            config['replaygain']['targetlevel'] = 89
```
Comment (diego-plan9) [inline test/test_replaygain.py]: This was needed as `GStreamerBackend.__init__` passes the config value to `_rg.set_property()` (which is present if run outside the test, as the plugin initialization takes care of creating the config value).

File: test/test_replaygain.py [hunk 2]
```diff
@@ -132,6 +141,20 @@ def test_cli_saves_album_gain_to_file(self):
...
+            GStreamerBackend(config['replaygain'], None)
+        except FatalReplayGainError as e:
+            # Skip the test if plugins could not be loaded.
+            if str(e) == "Failed to load required GStreamer plugins":
```
Comment (diego-plan9) [inline test/test_replaygain.py]: I have added an explicit check against the exception message, as `FatalReplayGainError` is used for several purposes. Might seem a bit redundant, but it is mainly for making the sure that the test is skipped only for the "missing plugins" reason, and it does not hide other errors in the future (as, at the moment, there seems to be no way to reach this line except for that reason).

**Comment by diego-plan9** [comment]
This is an attempt for fixing #1830 - I have manually tested with four setups:
- `gstreamer` installed, but not `plugins-*` -> the 3 tests skipped during `setUp()` 
- `gstreamer` and `plugins-base` installed, but not `plugins-good` or `plugins-bad`  -> the 3 tests skipped during `setUp()`
- `gstreamer`, `plugins-base` and `plugins-good` installed, but not `plugins-bad`  -> 2 tests executed and passed,  `test_cli_saves_track_gain` skipped during its execution.
- `gstreamer`, `plugins-base`, `plugins-good` and `plugins-bad` installed  -> 3 tests executed and passed.

File: test/test_replaygain.py (hunk 2 above)
Comment (sampsyo) [inline test/test_replaygain.py]: FWIW, one standard alternative to checking an exception string is to introduce a new exception subclass for exactly the case you're interested in.

**Comment by sampsyo** [comment]
Looks great to me! Thank you for the thorough testing—that looks like it took a lot of effort to set up. Nice work! :sparkles:

Merge at will.

File: test/test_replaygain.py (hunk 2 above)
Comment (diego-plan9) [inline test/test_replaygain.py]: Yep, that option felt more natural and clean but I did not want to mess with the Exception hierarchy as I wasn't really sure about the "exception branching policy" used in beets! I'll take the comment as a green light for subclassing at will, and make the pertinent changes.

Comment (sampsyo) [inline test/test_replaygain.py]: Sounds good. Yes, feel free to add a subclass—but I totally understand the instinct to use a light touch, too.

**Comment by diego-plan9** [comment]
Added a specific exception for the "gstreamer plugins could not be loaded", as discussed - so merging! :+1:

**Pull request merged**